docs/tutorial/creating_cli.md

```diff
@@ -0,0 +1,643 @@
+
+```scala 3
+//src/main/shellfish/contacts/core/ContactManager.scala
+override def searchName(name: Name): IO[List[Contact]] =
+  getAll.map(contacts =>
+    contacts.filter(c => c.firstName === name || c.lastName === name)
+  )
+```
+
+The first part is similar to the previous functions, we load all the contacts from the file, and then we filter the ones that contain the name (or field) we want to search. 
+
+
+The last function we need to implement is the `updateContact` function, be aware that this is the most complex function to implement, so we will break it down into smaller parts.
+
+```scala 3
+//src/main/shellfish/contacts/core/ContactManager.scala
+override def updateContact(
+    username: Username
+)(modify: Contact => Contact): IO[Contact] = 
+  for 
+    oldContact <- searchUsername(username) match // (1)
+      case None          => ContactNotFound(username).raiseError[IO, Contact]
+      case Some(contact) => contact.pure[IO]
+
+    updatedContacts = modify(oldContact) :: contacts.filterNot(_ == oldContact) // (2)
+    _ <- saveContacts(updatedContacts)
+  yield updatedContact
+      
+    
+```
+
+
+1. First, we'll find if the Contact we want to modify exists. If it doesnt, we raise a `ContactNotFound` error.
+
+2. In case the contact exists, we are going to apply to it the transform function and add it to the loaded list. While doing that, we delete the old contact via the `filterNot` method.
+
+3. Finally, we save the new contact list to the memory.
+
+Oof, that was a lot of code! But we're done with the core of our application. Now we can move to the CLI part.
+
+## CLI
+In this section, we will define the CLI part of our application. Here, we’re going to use the core logic previously implemented to interact with the CLI. For example, by printing to the standard output and reading from the user input.  
+
+It makes little sense to define the behaviour of the CLI in terms of an interface, as there are only a few ways of interacting with the user via a console. That why we will just create some functions contained on a `Cli` package:
+
+```scala 3
+//src/main/shellfish/contacts/cli/Cli.scala
+object Cli:
+  def addCommand ...
+```
+
+One way of passing the `ContactManager` dependency is to pass it as an argument to the function. However, it can be tedious to do this every time we want to use the functionality of our `Cli`, that's why Scala has an option to let the compiler do it for us! That's via the `given`/`using` clauses; if we want the compiler to "inject" the dependency, we just need to use the `using` keyword in the argument to the function, and it will be passed automatically as long as there is a given instance of our dependency in scope. This feature is also know as [contextual abstractions](https://docs.scala-lang.org/scala3/book/ca-context-parameters.html), We recommend that you take a closer look at these and how to use them in more detail.
```

Comment:
```suggestion
One way of passing the `ContactManager` dependency to these functions is to pass it as an argument. However, it can be tedious to do this every time we want to use the functionality of our `Cli`, that's why Scala has an option declare implicit paramaters via the `given`/`using` clauses. We just need to use the `using` keyword in front of any function's argument and as long as there is a given instance in scope at call site for every `using` clause, that value will be passed automatically. This feature is called [context parameters](https://docs.scala-lang.org/scala3/book/ca-context-parameters.html).
```